test/test_testing.py

```diff
@@ -683,5 +687,99 @@ def test_calculate_2_shards_against_optimal_shards(self):
                 self.assertEqual(sorted_tests, sorted_shard_tests)
 
 
+skip_if_no_cuda = unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
```

Comment:
This is probably not the right place for this. Maybe this could go directly into `torch.testing._internal.common_cuda`

Reply:
In other places this is spelled
```
from caffe2.python import workspace
@unittest.skipIf(not workspace.has_cuda_support, "No cuda support")
```